### PR TITLE
fix issue #20 and fix full screen padding

### PR DESCRIPTION
**Pull request by AnandJeyapal**
This is for the following issue 

https://github.com/dsa28s/compose-video/issues/20

### What was the issue?

Dialog window has extra padding around it. Hence if there is any content around the video player (top/bottom/right/left), `VideoPlayerFullScreenDialog` shows the content behind it.

### What is the fix?

We should update the layout properties of the dialog window so that it full occupies the screen's size.

I have copied the attributes from `activityWindow` and applied them to the `dialogWindow`.

Read more at: https://www.droidcon.com/2024/01/15/camouflage-the-status-bar-with-edge-to-edge-jetpack-compose-screens-and-dialogs/

### CODE REVIEW

**Comment by Less-dev** [comment]
cool bro

File: compose-video/src/main/kotlin/io/sanghun/compose/video/util/WindowUtil.kt [hunk 1]
```diff
@@ -33,15 +35,26 @@ internal fun Activity.setFullScreen(fullscreen: Boolean) {
...
-                or View.SYSTEM_UI_FLAG_FULLSCREEN
-            )
+        this.hideSystemBars()
     } else {
-        decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_VISIBLE
+        this.showSystemBars()
     }
 }
+
+private fun Window.hideSystemBars() {
```
Comment (dsa28s) [inline compose-video/src/main/kotlin/io/sanghun/compose/video/util/WindowUtil.kt:44]: Oh, awesome!

**Review by dsa28s — COMMENTED** [review]
Thanks for your contribution! I've written down some suggestions, please correct them.

File: compose-video/src/main/kotlin/io/sanghun/compose/video/VideoPlayerFullScreenDialog.kt [hunk 2]
```diff
@@ -77,6 +83,9 @@ internal fun VideoPlayerFullScreenDialog(
         PlayerView(context)
             .also(fullScreenPlayerView)
     }
+    var fullScreenModeEntered by remember {
```
[the remark this answers is not in the document]
Comment (AnandJeyapal) [reply compose-video/src/main/kotlin/io/sanghun/compose/video/VideoPlayerFullScreenDialog.kt]: Yes that makes sense. I have updated it :)

File: compose-video/src/main/kotlin/io/sanghun/compose/video/VideoPlayerFullScreenDialog.kt [hunk 3]
```diff
@@ -88,14 +97,31 @@ internal fun VideoPlayerFullScreenDialog(
...
 
-            (view.parent as DialogWindowProvider).window.setFullScreen(true)
+        val activityWindow = getActivityWindow()
+        val dialogWindow = getDialogWindow()
+
+        SideEffect {
+            if (activityWindow != null && dialogWindow != null && !fullScreenModeEntered) {
+                activityWindow.setFullScreen(true)
+                dialogWindow.setFullScreen(true)
+                dialogWindow
```
[the remark this answers is not in the document]
Comment (AnandJeyapal) [reply compose-video/src/main/kotlin/io/sanghun/compose/video/VideoPlayerFullScreenDialog.kt]: Thanks for the catch.  I removed it (also the redundant view variable).

File: compose-video/src/main/kotlin/io/sanghun/compose/video/VideoPlayerFullScreenDialog.kt [hunk 4]
```diff
@@ -139,3 +165,10 @@ internal fun VideoPlayerFullScreenDialog(
         }
     }
 }
+
+
+@Composable
+internal fun getDialogWindow(): Window? = (LocalView.current.parent as? DialogWindowProvider)?.window
+
+@Composable
+internal fun getActivityWindow(): Window? = LocalView.current.context.findActivity()?.window
```
[the remark this answers is not in the document]
Comment (AnandJeyapal) [reply compose-video/src/main/kotlin/io/sanghun/compose/video/VideoPlayerFullScreenDialog.kt]: My bad. I missed it. Thanks :)

**Review by dsa28s — APPROVED** [review]
Thank you!

**Comment by dsa28s** [comment]
This PR will release version 1.3.0

**Pull request merged**